### PR TITLE
Add a quick start guide

### PR DESCRIPTION
**Pull request by wozeparrot**
This adds a pretty simple quick start guide/tutorial that goes over the high level tinygrad API.

~~I do think it would be good to mention the JIT in this, probably just at the end as a note.~~ Added

I was aiming for this to be enough so that someone with zero knowledge of tinygrad should be able to learn enough to implement some larger models.
We do need more detailed docs tho.

Edit: ~~need to add saving and loading models as well but that depends on #882~~ Added

### CODE REVIEW

**Comment by geohot** [comment]
Great work! Maybe this is the thing that comes up when you click the docs folder and it links to the other two?

**Comment by wozeparrot** [comment]
I think it would be better if we had a sort of landing page for the docs, the current one is not great.

I'll throw something together.

**Comment by wozeparrot** [comment]
hmm, maybe I should just replace the repo readme as well, cuz what I have right now for the docs readme could also work for the repo readme with some tweaks and adding some missing stuff like the ops breakdown.

The current repo readme seems ok tho.

**Comment by wozeparrot** [comment]
I think this is in a state where we can get some comments on it.

You can see the rendered version here: https://github.com/wozeparrot/tinygrad/tree/tutorial/

This got a lot larger than I was intending for it to be but oh well we got nicer docs now ;)

**Comment by geohot** [comment]
I liked the copy and paste one-liner for lazy_matmul.

**Comment by wozeparrot** [comment]
> I like the copy and paste one-liner for lazy_matmul. I use it at a test all the time, and don't want to edit a file.

The main reason for splitting it up is that the one liner is a bit jarring for newcomers, as the readme will be the first thing that they see so it should be as well layed out as possible.

> I think the first thing in the README should be llama and stable diffusion (have to make sure the weight autodownloaders are good) Meh, maybe just surfacing the showcase readme is good and mention LLaMA and Stable Diffusion.

I do like the split out showcase as it makes the docs directory structure cleaner, and it lets me mention the showcase in more places. Yea I think if I just put a link to the showcase at the top and mention llama and stable diffusion it should be fine.

**Comment by geohot** [comment]
I need the one liner back

**Comment by wozeparrot** [comment]
It might also be good to mention mlperf somewhere here. Not sure tho

**Comment by geohot** [comment]
Big improvement, thanks!

File: docs/showcase.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+### Generative Adversarial Networks
+
+Take a look at [mnist_gan.py](/examples/mnist_gan.py).
+
+![mnist gan by tinygrad](/docs/showcase/mnist_by_tinygrad.jpg)
+
+### Stable Diffusion
+
+You will need to download the [weights](https://huggingface.co/CompVis/stable-diffusion-v-1-4-original/resolve/main/sd-v1-4.ckpt) of Stable Diffusion and put it into the [weights/](/weights) directory.
```
Comment (python273) [inline docs/showcase.md:44]: it's auto downloading them

File: docs/showcase.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+python3 examples/stable_diffusion.py
+```
+
+![a horse sized cat eating a bagel](/docs/showcase/stable_diffusion_by_tinygrad.jpg)
+
+*"a horse sized cat eating a bagel"*
+
+### LLaMA
+
+You will need to download and put the weights into the [weights/LLaMA](/weightsLLaMA) directory, which may need to be created.
```
Comment (python273) [inline docs/showcase.md:56]: typo in the link, though not sure if link is needed

**Pull request merged**